packages/grid/_modules_/grid/components/cell/GridEditSingleSelectCell.tsx

```diff
@@ -0,0 +1,77 @@
+import * as React from 'react';
+import Select, { SelectProps } from '@material-ui/core/Select';
+import MenuItem from '@material-ui/core/MenuItem';
+import { GridCellParams } from '../../models/params/gridCellParams';
+import {
+  GRID_CELL_EDIT_PROPS_CHANGE,
+  GRID_CELL_EDIT_PROPS_CHANGE_COMMITTED,
+  GRID_CELL_EDIT_EXIT,
+} from '../../constants/eventsConstants';
+import { isEscapeKey } from '../../utils/keyboardUtils';
+
+const renderSingleSelectOptions = (option) =>
+  typeof option === 'string' ? (
+    <MenuItem key={option} value={option}>
+      {option}
+    </MenuItem>
+  ) : (
+    <MenuItem key={option.value} value={option.value}>
+      {option.label}
+    </MenuItem>
+  );
+
+export function GridEditSingleSelectCell(props: GridCellParams & SelectProps) {
+  const {
+    id,
+    value,
+    formattedValue,
+    api,
+    field,
+    row,
+    colDef,
+    cellMode,
+    isEditable,
+    className,
+    getValue,
+    hasFocus,
+    ...other
+  } = props;
+
+  const handleChange = (event) => {
+    const editProps = { value: event.target.value };
+
+    if (event.key) {
+      api.publishEvent(GRID_CELL_EDIT_PROPS_CHANGE, { id, field, props: editProps }, event);
+    } else {
+      api.publishEvent(
+        GRID_CELL_EDIT_PROPS_CHANGE_COMMITTED,
+        { id, field, props: editProps },
+        event,
+      );
+      api.publishEvent(GRID_CELL_EDIT_EXIT, { id, field }, event);
+    }
+  };
+
+  const handleClose = (event, reason) => {
+    if (reason === 'backdropClick' || isEscapeKey(event.key)) {
+      api.publishEvent(GRID_CELL_EDIT_EXIT, { id, field });
```

Comment:
Better to use our own APIs.

```suggestion
      api.setCellMode(id, field, 'view');
```

Reply:
This is different than what we've used for now. I don't remember discussing that pattern. The API is only called directly in the data generator package.

Reply:
In #1955 I changed the other components to use the same approach. The imperative API is better because we can enforce the type of the params.

The demo about custom components is already using the API methods: https://material-ui.com/components/data-grid/editing/#custom-edit-component